common/djangoapps/third_party_auth/saml.py

```diff
@@ -35,14 +33,18 @@ def setting(self, name, default=None):
 
     def auth_url(self):
         """
-        Check that the request includes an 'idp' parameter before getting the
-        URL to which we must redirect in order to authenticate the user.
+        Check that SAML is enabled and that the request includes an 'idp'
+        parameter before getting the URL to which we must redirect in order to
+        authenticate the user.
 
+        raise Http404 if SAML is disabled
         raise AuthMissingParameter if the 'idp' parameter is missing.
 
         TODO: remove this method once the fix is merged upstream:
         https://github.com/omab/python-social-auth/pull/821
         """
+        if not self._config.enabled:
```

Comment:
This code differs from the upstream version, and we want to remove this whole method once the upstream patch has been merged. With the change I suggested above, we can remove these two lines, though the result is a 301 redirect instead of a 404 error. Although that's not as good as a 404, I think it may be better in this case since it reduces the difference with upstream. What do you think?